pandas/core/indexes/base.py

```diff
@@ -7538,6 +7539,12 @@ def _unpack_nested_dtype(other: Index) -> Index:
         # If there is ever a SparseIndex, this could get dispatched
         #  here too.
         return dtype.categories
+    elif isinstance(dtype, ArrowDtype):
+        # GH 53617
+        import pyarrow as pa
+
+        if pa.types.is_dictionary(dtype.pyarrow_dtype):
+            other = other.astype(ArrowDtype(dtype.pyarrow_dtype.value_type))
```

Comment:
this is going to have implications beyond the constructors

Reply:
Added some tests for `get_indexer`, `get_indexer_non_unique` that seem related. LMK if there are other methods you also had in mind

Reply:
new tests look good, thanks.

I think we can make this a lot cheaper though since the places that all _unpack_nested_dtype only need the result's dtype.  so with a little tinkering we can just return that and avoid doing a cast

Reply:
Makes sense. Might be better as a followup since this change is being backported?

Reply:
sure